enterprise/api/v1/views.py

```diff
@@ -525,6 +526,39 @@ def unlink_users(self, request, pk=None):  # pylint: disable=unused-argument
 
         return Response(status=HTTP_200_OK)
 
+    @action(detail=False)
+    def algolia_key(self, request, *args, **kwargs):
+        """
+        Returns an Algolia API key that is secured to only allow searching for
+        objects associated with enterprise customers that the user is linked to.
+        """
+
+        if not settings.ENTERPRISE_ALGOLIA_SEARCH_API_KEY:
+            return Response("Algolia API key not configured", status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
```suggestion
            LOGGER.warning("Algolia search API key is not configured. To enable this view, "
                           "set `ENTERPRISE_ALGOLIA_SEARCH_API_KEY` in settings.")
            raise Http404
```

Why do we want to raise 500? We could treat this endpoint as non-existent if the Algolia key is not configured.
Also, is there any value in showing the reason for this behavior to the users instead of showing a warning in the logs?

Reply:
@Agrendalath, Django throws [ImproperlyConfigured](https://docs.djangoproject.com/en/4.2/ref/exceptions/#improperlyconfigured) exception when something is wrong with the settings, so I though 500 here would be consistent with the behavior. But after re-reading the description, I agree that 404 is better, as unset `ENTERPRISE_ALGOLIA_SEARCH_API_KEY` is one of the expected options.